### ISSUE TITLE
Being able to navigate while writing a mail

### ISSUE THREAD

**Issue by GretaD** [nextcloud/mail#2831]
### Feature Request

As a Nextcloud Mail user I want to navigate the app while the composer is open.

### Solution

Move the composer into some sort of popup.

~~To be able to compose in a new window. Just like other mail apps, would be nice to have this feature. Its more conformable to have your composer in the middle of the screen, without fearing you would accidentally click a message while you write. Also its less distracting.~~

**Comment by jancborchardt** [comment]
@GretaD do you want to analyze how for example Gmail, Microsoft365 and Apple Mail do this and post screenshots here so we can have that as a reference for designing it?

**Comment by GretaD** [comment]
Yes, i can analyze gmail.

**Comment by jancborchardt** [comment]
What’s definitely nice in Gmail is the ability to compose and still being able to navigate and read other mails. We could think about doing this pane-docked-to-bottom as they do? This is the default, and then you can either minimize it or fullscreen it.

![](https://storage.googleapis.com/gweb-uniblog-publish-prod/original_images/Smart-compose_Taco_Tuesday.gif)

**Comment by ChristophWurst** [comment]
Even better: if you have this, you could embed Mail into other apps' pages for mailto links and such

**Comment by GretaD** [comment]
Yes, gmail is the best option so far.

Outlook, has 2 steps to open in a new tab, when you click "New Message" it opens the composer on the third column, just like we have it now. And then if you open in a new window it looks like this:
![outlook2](https://user-images.githubusercontent.com/12728974/78265211-6e5b0800-7504-11ea-8e04-acb183e29110.png)

Apple(from internet) more or less like outlook.
![applemail](https://user-images.githubusercontent.com/12728974/78265367-a2362d80-7504-11ea-8eff-60e9b73c2b97.jpg)

**Comment by jancborchardt** [comment]
> Even better: if you have this, you could embed Mail into other apps' pages for mailto links and such

Yup, this would be a killer integration and we wouldn’t have to rip people out of the flow. We should then also make it the default just like Gmail does, so there’s no different modes.

**Comment by DrRSatzteil** [comment]
Ideally I think this should be made available to other apps as well. This would be very useful for talk for example and it would definitely make sense to have a similar look when it is used with other apps. So in my opinion it would make sense to have a more generic approach for this component to make it reusable in other contexts.

**Comment by ChristophWurst** [comment]
@jancborchardt how can we achieve the design from https://github.com/nextcloud/mail/issues/2831#issuecomment-607851153 in Nextcloud? Would you suggest using any of the existing components, like is this similar to anything that we already have, or do we need a new one for this?

**Comment by jancborchardt** [comment]
> @jancborchardt how can we achieve the design from [#2831 (comment)](https://github.com/nextcloud/mail/issues/2831#issuecomment-607851153) in Nextcloud? Would you suggest using any of the existing components, like is this similar to anything that we already have, or do we need a new one for this?

It seems it should be a new component. I can’t think of one which is similar. The issue though is that we need to make sure this works with other apps, e.g. when you collapse it, it should still show a header part in the bottom right of the screen – however in apps like Talk, when you have a call, there’s important stuff in the bottom right like your own call controls as well as the sidebar chat input box.

So possibly doing it like Outlook as @GretaD posted in https://github.com/nextcloud/mail/issues/2831#issuecomment-607904850 is a good intermediate step – we could just show the composer in a popup?

**Comment by ChristophWurst** [comment]
Sounds good.

In my understanding the docked mode would be the default if we have it. The problems in other apps are only there once we actually show Mail on other pages, so that is something for later and not important to this discussion right now.

The popup would be only optional and the default composer would still remain inside the main tab, right? Should there be an option to "move" the composer to a popup?

> And then if you open in a new window it looks like this

@GretaD do they have a button for that?

**Comment by GretaD** [comment]
> 
> > And then if you open in a new window it looks like this
> 
> @GretaD do they have a button for that?

![photo5807729012322186484](https://user-images.githubusercontent.com/12728974/106117050-ec9f2f80-6152-11eb-9b00-5148b704c2d9.jpg)

They have an icon Top Right.

**Comment by jancborchardt** [comment]
> The popup would be only optional and the default composer would still remain inside the main tab, right? Should there be an option to "move" the composer to a popup?

Yeah, that should most likely go into the 3-dot menu next to the "Reply"/"Send" button.

**Comment by handkerchief333** [comment]
Ok great, I didn't see that there was already a feature request. Now that it's almost 4 years old, is something concrete planned in this direction or what's holding it up?

**Comment by szimszon** [comment]
Any progress?

**Comment by handkerchief333** [comment]
@ChristophWurst Okay, but the minimize function isn't exactly what some people want. It would be great if you could do both at the same time. In other words, have another email open while you're writing a new message. 

Otherwise, you always have to minimize, check, maximize, continue writing, minimize again, check again, and so on. To do this, I currently have to run the mail app in two different browser tabs. In Gmail, you can compose another email while writing an email without switching back and forth. It saves time and frustration. If you want, I can open a separate feature request for this.
